js/modules/k6/ws/ws_test.go

```diff
@@ -240,13 +274,18 @@ func TestSession(t *testing.T) {
 			}, 0);
 		});
 		`))
-		require.Error(t, err)
-		require.Contains(t, err.Error(), "setTimeout requires a >0 timeout parameter, received 0.00 ")
-		assertSessionMetricsEmitted(t, metrics.GetBufferedSamples(test.samples), "", sr("WSBIN_URL/ws-echo"), statusProtocolSwitch, "")
-	})
+	require.ErrorContains(t, err, "setTimeout requires a >0 timeout parameter, received 0.00 ")
+	// TODO SessionDuration is not emitted if the "open" handler returns an error - which is what happens here
+	// assertSessionMetricsEmitted(t, metrics.GetBufferedSamples(test.samples), "", sr("WSBIN_URL/ws-echo"), statusProtocolSwitch, "")
```

Comment:
This took me a while to figure out, but the thing was that the previous way the test were used - samples kept accumulating and as the same URL was used, later tests could see samples from *older* tests :facepalm: 

I could add `SessionDuration` here by moving the `defer` that emits it above the execution of the "open" handler? 

But didn't want to do a functioncal change with no discussion. 

It does make some sense that if the connection is opened we will emit it, and this likely should've been the case, and given the test maybe the expectation :shrug:

Reply:
I've opened https://github.com/grafana/k6/issues/2734 for this

Reply:
Could you please also refer to the issue in the TODO comment?

Reply:
```suggestion
	// TODO SessionDuration is not emitted if the "open" handler returns an error - which is what happens here
	// see https://github.com/grafana/k6/issues/2734
	// assertSessionMetricsEmitted(t, metrics.GetBufferedSamples(test.samples), "", sr("WSBIN_URL/ws-echo"), statusProtocolSwitch, "")
```

@olegbespalov ?

Reply:
Looks good to me! :+1: